frontend/src/components/meetTheWhales/MeetTheWhalesPage.tsx

```diff
@@ -1,38 +1,103 @@
-import { useEffect, useState } from "react";
+import React, { useState, FormEvent, useEffect, useContext } from "react";
 import "./MeetTheWhalesPage.scss";
-import { fetchSpecies, Species } from "../../clients/apiClients";
+import {
+  Whales,
+  GetAllWhales,
+  createInteraction,
+  User,
+} from "../../clients/apiClients";
+import { LoginContext } from "../login/LoginManager";
+
+type FromStatus = "READY" | "SUBMITTING" | "ERROR" | "FINISHED";
 
 export function MeetTheWhalesPage(): JSX.Element {
-  const [species, setSpecies] = useState<Array<Species>>([]);
+  const [whales, setWhales] = useState<Array<Whales>>([]);
+  const [date, setDate] = useState<Date>(new Date());
+  const [whaleId, setWhaleId] = useState<number>(1);
+  const [status, setStatus] = useState<FromStatus>("READY");
+
+  const { username, password } = useContext(LoginContext);
+  const loginContext = useContext(LoginContext);
+
+  const submitForm = (event: FormEvent) => {
+    event.preventDefault();
+    setStatus("SUBMITTING");
+    createInteraction(
+      {
+        date,
+        whaleId,
+      },
+      username,
+      password
+    )
+      .then(() => {
+        setStatus("FINISHED");
+        GetAllWhales().then(setWhales);
+      })
+      .catch(() => setStatus("ERROR"));
+  };
+
+  function stringOfUsers(arr: User[]) {
+    if (arr.length > 0) {
+      return arr.slice(1).reduce(function (acc, currVal) {
+        return acc + (", " + currVal.name + " (" + currVal.username + ") ");
+      }, " " + arr[0].name + " (" + arr[0].username + ")");
+    } else return "";
+  }
+
+  function AlreadySponsor(name: string, arr: User[]) {
```

Comment:
Functions that aren't React components should be lowercase
```suggestion
  function alreadySponsor(name: string, arr: User[]) {
```